V2_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,183 @@
+    logoutPath: '/custom/logout',
+    redirectUriPath: '/custom/callback',
+    postLogoutRedirectUri: '/custom/post-logout',
+  })
+);
+
+// After
+app.use(
+  auth({
+    routes: {
+      login: '/custom/login',
+      logout: '/custom/logout',
+      callback: '/custom/callback',
+      postLogoutRedirect: '/custom/post-logout',
+    },
+  })
+);
+```
+
+### Session Lifecycle configuration
+
+Session duration was rolling only and configured using `appSession.duration` (default 24hrs). Now it can be configured to rolling or absolute (default rolling 24hrs and absolute 7 days).
+
+```js
+// Before
+app.use(
+  auth({
+    appSession: {
+      duration: 86400, // default 1 day in secs
+    },
+  })
+);
+
+// After
+app.use(
+  auth({
+    appSession: {
+      rolling: true,
+      rollingDuration: 86400, // default 1 day rolling duration in secs
+      absoluteDuration: 86400 * 7, // default 7 days absolute duration in secs
+    },
+  })
+);
+```
+
+### `required` is now `authRequired`
+
+To enable or disable all routes to require authentication, use the `authRequired` configuration (default `true`)
+
+```js
+// Before
+app.use(
+  auth({
+    required: true,
+  })
+);
+
+// After
+app.use(
+  auth({
+    authRequired: true,
+  })
+);
+```
+
+### Configuration items renamed/removed
+
+- **`required` is now `authRequired`** - to enable or disable all routes to require authentication, use the `authRequired` configuration (default `true`)
+- **`idTokenAlg` is now `idTokenSigningAlg`** - to specify an id token signing algorithm, use `idTokenSigningAlg`
+- **`httpOptions`** - it is no longer possible to pass custom http options to the underlying library
+- **`handleCallback`** and **`getUser`** - The hooks have been removed and will be replaced by a collection of hooks in a later release.
+
+```js
+// Before
+app.use(
+  auth({
+    required: true,
+    idTokenAlg: 'RS256',
+    httpOptions: {},
+    handleCallback: () => {},
+    getUser: () => {},
+  })
+);
+
+// After
+app.use(
+  auth({
+    authRequired: true,
+    idTokenSigningAlg: 'RS256',
+  })
+);
+```
+
+## Session and Context
+
+This library adds an auth context to the request and response objects used within route handling called `oidc` (was `openid`).
+
+```js
+// Before
+app.get('/', (req, res) => {
+  const user = req.openid.user;
+  const client = req.openid.client; // It is no longer possible to access the underlying client
+  const isAuthenticated = req.isAuthenticated();
+  const tokenSet = req.makeTokenSet({ tokens });
+
+  // Login
+  res.openid.login({});
+  // Logout
+  res.openid.logout({});
+});
+
+// After
+app.get('/', async (req, res) => {
+  const user = req.oidc.user;
+  const claims = req.oidc.idTokenClaims;
+  const isAuthenticated = req.oidc.isAuthenticated();
+  const idToken = req.oidc.idToken;
+  const {
+    access_token,
+    token_type,
+    expires_in,
+    isExpired,
+    refresh,
+  } = req.oidc.accessToken; // If `code` in response_type
```

Comment:
this is just an observation, but the property `oidc.accessToken` has an `access_token` property inside. Looks like it refers to the raw value of the token. In that case, if I want to access the access token value, it looks like I need to do `req.oidc.accessToken.access_token` 😕  shouldn't that be renamed to `raw` or `value`?

Also, why the `idToken` shares the same level with the `idTokenClaims` property, what's the reason to have it different from the access token?

Reply:
I'm happy with the API, but let me check to be safe

@panva @vibronet wdyt?

Reply:
The reason why `accessToken` (named camelCase) is an object is that an Access Token is defined by more than just its value. It has a type, a possible expiration and optionally can be refreshed. That is why it's all "namespaced". I would be okay with renaming `accessToken.access_token` to `accessToken.value` if it makes sense to others. As is it has the structure of an OAuth Access Token Response - `{ access_token, expires_in, token_type }` with additional helper functions slapped in there.

I am indifferent to how `idToken` (holds the raw value) and `idTokenClaims` are exposed.

Reply:
ok - I'll check this with @vibronet and raise some additional work before the Beta launch if we want to change it

Reply:
Interesting, good catch @lbalmaceda - `accessToken.access_token` does look odd. Renaming to `value` works for me, as long as we are consistent across the API surface (e.g. `idToken` is also a container, exposing `claims` and `value`)